pkg/collector/corechecks/snmp/internal/profile/profile_resolver.go

```diff
@@ -19,40 +20,37 @@ import (
 
 func resolveProfiles(userProfiles, defaultProfiles ProfileConfigMap) (ProfileConfigMap, error) {
 	rawProfiles := mergeProfiles(defaultProfiles, userProfiles)
-
 	userExpandedProfiles, err := loadResolveProfiles(rawProfiles, defaultProfiles)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load profiles: %s", err)
 	}
-	profiles := ProfileConfigMap{}
-
-	for key, val := range userExpandedProfiles {
-		if strings.HasPrefix(key, "_") {
-			continue
-		}
-		profiles[key] = val
-	}
-	return profiles, nil
+	return userExpandedProfiles, nil
 }
 
 func loadResolveProfiles(pConfig ProfileConfigMap, defaultProfiles ProfileConfigMap) (ProfileConfigMap, error) {
 	profiles := make(ProfileConfigMap, len(pConfig))
 
-	for name, profConfig := range pConfig {
-		err := recursivelyExpandBaseProfiles(name, &profConfig.Definition, profConfig.Definition.Extends, []string{}, pConfig, defaultProfiles)
+	for name := range pConfig {
+		// No need to resolve abstract profile
+		if strings.HasPrefix(name, "_") {
+			continue
+		}
+
+		newProfileConfig := deepcopy.Copy(pConfig[name]).(ProfileConfig)
```

Comment:
Note for reviewers: this deepcopy will avoid modification of the original profile.